syntax = "proto3";

package kvstore.client;

option cc_enable_arenas = true;

message GetKey {
  uint32 key = 1;
}

message GetKeyResp {
  uint32 stream = 1;
  string val = 2;
}

message PutKey {
  uint32 key = 1;
  string val = 2;
  uint32 consistency = 3;
}

message ClientMessage {
  oneof msg {
    GetKey get_key = 1;
    PutKey put_key = 2;
  }
}
